### PR TITLE
Rename action improvements and compatible IDEs update

### PR DESCRIPTION
**Pull request by VladRassokhin**
Hopefully commit messages are self-explanatory.

### CODE REVIEW

**Comment by VladRassokhin** [comment]
I've run previous plugin version verification against 231 EAP and [it passed](https://plugins.jetbrains.com/plugin/17063-regex-rename-files/versions/stable/263888).

File: src/main/kotlin/nl/bryanderidder/regexrenamefiles/ReplaceFileNamesAction.kt [hunk 1]
```diff
@@ -2,70 +2,114 @@ package nl.bryanderidder.regexrenamefiles
...
+                errors.add("File '${it.file.name}' is not valid one")
+                continue
+            }
+            if (!it.file.fileSystem.isValidName(it.newName)) {
+                errors.add("New file name '${it.newName}' is not supported by file system")
+                continue
+            }
+            val parent = it.file.parent
+            if (parent?.findChild(it.newName) != null) {
+                errors.add("New name '${it.newName}' for '${it.oldName}' already used in directory '${parent.path}'")
```
Comment (Bryanx) [inline src/main/kotlin/nl/bryanderidder/regexrenamefiles/ReplaceFileNamesAction.kt]:
```suggestion
                errors.add("New name '${it.newName}' for '${it.oldName}' is already used in directory '${parent.path}'")
```

File: src/main/kotlin/nl/bryanderidder/regexrenamefiles/ReplaceFileNamesAction.kt [hunk 2]
```diff
@@ -2,70 +2,114 @@ package nl.bryanderidder.regexrenamefiles
...
-            invokeLater {
-                showErrorDialog(project, errorMessage, dialog.title)
+        }
+        for (it in events) {
+            if (!it.file.isValid) {
+                errors.add("File '${it.file.name}' is not valid one")
+                continue
+            }
+            if (!it.file.fileSystem.isValidName(it.newName)) {
+                errors.add("New file name '${it.newName}' is not supported by file system")
```
Comment (Bryanx) [inline src/main/kotlin/nl/bryanderidder/regexrenamefiles/ReplaceFileNamesAction.kt]:
```suggestion
                errors.add("New file name '${it.newName}' is not supported by the file system")
```

File: src/main/kotlin/nl/bryanderidder/regexrenamefiles/ReplaceFileNamesAction.kt [hunk 3]
```diff
@@ -2,70 +2,114 @@ package nl.bryanderidder.regexrenamefiles
...
-            val errorMessage = "File '${renameEvent.newName}' already exists " +
-                    "in directory '${renameEvent.file.parent.path}'"
-            LOGGER.warn(errorMessage, e)
-            revertRenameFiles()
-            invokeLater {
-                showErrorDialog(project, errorMessage, dialog.title)
+        }
+        for (it in events) {
+            if (!it.file.isValid) {
+                errors.add("File '${it.file.name}' is not valid one")
```
Comment (Bryanx) [inline src/main/kotlin/nl/bryanderidder/regexrenamefiles/ReplaceFileNamesAction.kt]:
```suggestion
                errors.add("File '${it.file.name}' is not valid")
```

**Review by Bryanx — COMMENTED** [review]
Thank you for the PR. Great improvements to the error handling. I quickly checked and saw some error messages that could be improved, other than that looks good. I will test this locally tomorrow and make a new release if everything is fine.

**Review by Bryanx — APPROVED** [review]

**Comment by VladRassokhin** [comment]
Thank you for fast review and merge!

**Pull request merged**